extension/persistence/eclipselink/src/main/java/org/apache/polaris/extension/persistence/impl/eclipselink/PolarisSequenceUtil.java

```diff
@@ -82,20 +82,29 @@ public static void initialize(EntityManager session) {
           long queryResult =
               (long) session.createNativeQuery("SELECT NEXTVAL('POLARIS_SEQ')").getSingleResult();
           result = Optional.of(queryResult);
+          LOGGER.info("query result " + queryResult);
         } else {
           LOGGER.info("POLARIS_SEQ does not exist, skipping NEXTVAL");
         }
         result.ifPresent(
             r -> {
               ModelSequenceId modelSequenceId = new ModelSequenceId();
               modelSequenceId.setId(r);
-
+              EntityTransaction transaction = session.getTransaction();
               // Persist the new ID:
-              session.persist(modelSequenceId);
-              session.flush();
 
-              // Clean the sequence:
-              removeSequence(session);
+              try {
+                transaction.begin();
+                session.persist(modelSequenceId);
+                session.flush(); //
+                removeSequence(session);
+                transaction.commit();
```

Comment:
Can you explain more about this change, and how it interacts with an existing txn?